src/components/navBar/navBar.js

```diff
@@ -469,6 +470,10 @@ function MdNavItem($mdAria, $$rAF) {
           scope.$apply();
         });
 
+          attrs.$observe('disabled', function (value) {
```

Comment:
This seems unnecessary since ngDisabled does this automatically for us. It's built into AngularJS already. I don't think think we need to touch any of the JS at all.

https://docs.angularjs.org/api/ng/directive/ngDisabled

Reply:
I didn't quite understand what you mean, how can we disable the inner button element?

Reply:
Ah, I see what you mean. 

We have to be careful with adding features like this. The problem is introducing these checks will add on to the digest loop with watchers/observers. This would cause an unexpected strain on the CPU when users upgrade, especially on something like nav-item.

Using a MutationObserver on the `ctrl` element's `disabled` attribute would be more performant. If the browser doesn't support it, then we can fall back to watchers/observers.

Reply:
@currensy are you still able to look into this?

I know that there aren't many examples of doing this in the AngularJS Material code base, but here is one: [tabsDummyWrapperDirective.js](https://github.com/angular/material/blob/master/src/components/tabs/js/tabsDummyWrapperDirective.js#L24-L52).

It's not an exact match for what needs to be done here, but hopefully it can put you on the right track?

Reply:
```javascript
if('MutationObserver' in $window) {
          var config = {attributes: true, attributeFilter: ['disabled']};
          var targetNode = element[0];
          var mutationCallback = function(mutationList) {
            mdNavItem.disabled = $mdUtil.parseAttributeBoolean(attrs[mutationList[0].attributeName], false);
            scope.$apply();
          };
          var observer = new MutationObserver(mutationCallback);
          observer.observe(targetNode, config);
          disconnect = observer.disconnect.bind(observer);
        } else {
          attrs.$observe('disabled', function (value) {
            mdNavItem.disabled = $mdUtil.parseAttributeBoolean(value, false);
          });
        }
```
-------------
The demo seems to be working fine but navBar's disable support tests are failing. Did I miss something?

Reply:
What do the test failures look like?

Reply:
Expected undefined to be true.
	@src/components/navBar/navBar.spec.js:190:11

Expected undefined to be false.
	@src/components/navBar/navBar.spec.js:204:11
	Expected undefined to be true.
	@src/components/navBar/navBar.spec.js:207:11

Reply:
You may also want to refer to the use of MutationObserver in [tooltip.js](https://github.com/angular/material/blob/master/src/components/tooltip/tooltip.js#L157-L193).

In your code, I'm not sure that `mutationList[0]` is always the right mutation? In the tooltip example it goes through all of the mutations looking for `mutation.attributeName === 'disabled'`.

Reply:
Should the target node just be `element` instead of `element[0]`?

Reply:
I guess the attributeFilter is supposed to ensure that? - Exactly.

I tried to filter it manually anyway, but the result stays the same. It looks like the evaluation in the test happens prior to the mdNavItem.disabled assignment, some sort of race condition...

Reply:
element is a wrapped node while the observer expects to get [Node](https://developer.mozilla.org/en-US/docs/Web/API/Node)

Reply:
Maybe instead of `scope.$apply();`?
```
        $mdUtil.nextTick(function() {
          mdNavItem.disabled = $mdUtil.parseAttributeBoolean(attrs[mutationList[0].attributeName], false);
        });
```

Reply:
I've actually made this change (looks better than calling $apply imo) - no luck with tests though

Reply:
Ah yeah, it wouldn't want the jQuery wrapped element. If you can't figure it out, you can push up your changes and I can try to debug it in the next few days.